lib/chef/knife/ssh.rb

```diff
@@ -247,19 +247,20 @@ def search_nodes
       # @since 12.5.0
       # @param host [String] Hostname for this session.
       # @param port [String] SSH port for this session.
+      # @param type [Integer] Flag: host or gateway key
       # @param user [String] Optional username for this session.
       # @return [Hash<Symbol, Object>]
-      def session_options(host, port, user = nil)
+      def session_options(host, port, type = 0, user = nil)
```

Comment:
this should be a named parameter:

```
def session_options(host, port, user = nil, gateway = false)
```

then there's no need to pass it `false` in the normal host case, but the gateway would call it like:

```ruby
session_options(gw_host, gw_port, gw_user, gateway: true)
```

which makes it a lot more obvious what is happening in that line than having a strange '1' as an option.